scripts/ci_tests/rust_tests.sh

```diff
@@ -186,39 +270,79 @@ function restart_postgres() {
+        cd ${ROOT}/dev-tools/pg-services-local
+        docker-compose down -v postgres
+        docker-compose up -d postgres
+        PGPASSWORD=$POSTGRES_PASSWORD psql -h localhost -U $POSTGRES_USER -c 'CREATE DATABASE IF NOT EXISTS iota_indexer;' -c 'ALTER SYSTEM SET max_connections = 500;' 2>/dev/null
+    )
 }
 
 function tests_using_postgres() {
-    if [ "$RESTART_POSTGRES" == "true" ]; then restart_postgres; fi
-    cargo nextest run --no-fail-fast --test-threads 1 --package iota-graphql-rpc --test e2e_tests --test examples_validation_tests --features pg_integration
-    cargo nextest run --no-fail-fast --test-threads 1 --package iota-graphql-rpc --lib --features pg_integration -- test_query_cost
-    cargo nextest run --no-fail-fast --test-threads 8 --package iota-graphql-e2e-tests --features pg_integration
-    cargo nextest run --no-fail-fast --test-threads 1 --package iota-cluster-test --test local_cluster_test --features pg_integration
-    cargo nextest run --no-fail-fast --test-threads 1 --package iota-indexer --test ingestion_tests --features pg_integration
-    # Iota-indexer's RPC tests, which depend on a shared runtime, are incompatible with nextest due to its process-per-test execution model.
-    # cargo test, on the other hand, allows tests to share state and resources by default.
-    cargo test --profile simulator --package iota-indexer --test rpc-tests --features shared_test_runtime
+    # we run this in a subshell to avoid polluting the environment with the variables set in this function
+    (
+        # Tests written with #[sim_test] are often flaky if run as #[tokio::test] - this var
+        # causes #[sim_test] to only run under the deterministic `simtest` job, and not the
+        # non-deterministic `test` job.
+        export IOTA_SKIP_SIMTESTS=1
+
+        if [ "$RESTART_POSTGRES" == "true" ]; then restart_postgres; fi
+        print_and_run_command "cargo nextest run --config-file .config/nextest.toml --profile threads-1 --package iota-graphql-rpc --test e2e_tests --test examples_validation_tests --features pg_integration ${ENABLE_NO_CAPTURE:+--nocapture}"
+        print_and_run_command "cargo nextest run --config-file .config/nextest.toml --profile threads-1 --package iota-graphql-rpc --lib --features pg_integration -- test_query_cost ${ENABLE_NO_CAPTURE:+--nocapture}"
+        print_and_run_command "cargo nextest run --config-file .config/nextest.toml --profile threads-8 --package iota-graphql-e2e-tests --features pg_integration ${ENABLE_NO_CAPTURE:+--nocapture}"
+        print_and_run_command "cargo nextest run --config-file .config/nextest.toml --profile threads-1 --package iota-cluster-test --test local_cluster_test --features pg_integration ${ENABLE_NO_CAPTURE:+--nocapture}"
+        print_and_run_command "cargo nextest run --config-file .config/nextest.toml --profile threads-1 --package iota-indexer --test ingestion_tests --features pg_integration ${ENABLE_NO_CAPTURE:+--nocapture}"
```

Comment:
Hmm that didn't help much did it 😆 maybe instead of profiles we could have a common base for the command?

```sh
NEXTEST_RUN="cargo nextest run --config-file .config/nextest.toml --features pg_integration"

print_and_run_command "$NEXTEST_RUN -p iota-graphql-rpc --test-threads 1 --test e2e_tests --test examples_validation_tests ${ENABLE_NO_CAPTURE:+--nocapture}"
...
```